pkg/parse/root.go

```diff
@@ -381,6 +381,140 @@ func setRenderingStatusFields(rendering *v1beta1.RenderingStatus, p Parser, newS
+	var sourceSpec, renderSpec, syncSpec SourceSpec
+	switch sourceType {
+	case configsync.GitSource:
+		if rsyncStatus.Source.Git != nil {
+			sourceSpec = GitSourceSpec{
+				Repo:     rsyncStatus.Source.Git.Repo,
+				Revision: rsyncStatus.Source.Git.Revision,
+				Branch:   rsyncStatus.Source.Git.Branch,
+				Dir:      rsyncStatus.Source.Git.Dir,
+			}
+		}
+		if rsyncStatus.Rendering.Git != nil {
+			renderSpec = GitSourceSpec{
+				Repo:     rsyncStatus.Rendering.Git.Repo,
+				Revision: rsyncStatus.Rendering.Git.Revision,
+				Branch:   rsyncStatus.Rendering.Git.Branch,
+				Dir:      rsyncStatus.Rendering.Git.Dir,
+			}
+		}
+		if rsyncStatus.Sync.Git != nil {
+			syncSpec = GitSourceSpec{
+				Repo:     rsyncStatus.Sync.Git.Repo,
+				Revision: rsyncStatus.Sync.Git.Revision,
+				Branch:   rsyncStatus.Sync.Git.Branch,
+				Dir:      rsyncStatus.Sync.Git.Dir,
+			}
+		}
+	case configsync.OciSource:
+		if rsyncStatus.Source.Oci != nil {
+			sourceSpec = OCISourceSpec{
+				Image: rsyncStatus.Source.Oci.Image,
+				Dir:   rsyncStatus.Source.Oci.Dir,
+			}
+		}
+		if rsyncStatus.Rendering.Oci != nil {
+			renderSpec = OCISourceSpec{
+				Image: rsyncStatus.Rendering.Oci.Image,
+				Dir:   rsyncStatus.Rendering.Oci.Dir,
+			}
+		}
+		if rsyncStatus.Sync.Oci != nil {
+			syncSpec = OCISourceSpec{
+				Image: rsyncStatus.Sync.Oci.Image,
+				Dir:   rsyncStatus.Sync.Oci.Dir,
+			}
+		}
+	case configsync.HelmSource:
+		if rsyncStatus.Source.Helm != nil {
+			sourceSpec = HelmSourceSpec{
+				Repo:    rsyncStatus.Source.Helm.Repo,
+				Chart:   rsyncStatus.Source.Helm.Chart,
+				Version: rsyncStatus.Source.Helm.Version,
+			}
+		}
+		if rsyncStatus.Rendering.Helm != nil {
+			renderSpec = HelmSourceSpec{
+				Repo:    rsyncStatus.Rendering.Helm.Repo,
+				Chart:   rsyncStatus.Rendering.Helm.Chart,
+				Version: rsyncStatus.Rendering.Helm.Version,
+			}
+		}
+		if rsyncStatus.Sync.Helm != nil {
+			syncSpec = HelmSourceSpec{
+				Repo:    rsyncStatus.Sync.Helm.Repo,
+				Chart:   rsyncStatus.Sync.Helm.Chart,
+				Version: rsyncStatus.Sync.Helm.Version,
+			}
+		}
+	}
+
+	return &ReconcilerStatus{
+		SourceStatus: &SourceStatus{
+			Spec:   sourceSpec,
+			Commit: rsyncStatus.Source.Commit,
+			// Can't parse errors.
+			// Errors will be reset the next time the reconciler updates the status.
+			Errs:       nil,
+			LastUpdate: rsyncStatus.Source.LastUpdate,
+		},
+		RenderingStatus: &RenderingStatus{
+			Spec:    renderSpec,
+			Commit:  rsyncStatus.Rendering.Commit,
+			Message: rsyncStatus.Rendering.Message,
+			// Can't parse errors.
+			// Errors will be reset the next time the reconciler updates the status.
+			Errs:       nil,
+			LastUpdate: rsyncStatus.Rendering.LastUpdate,
+			// Whether there exists a kustomize.yaml in the source is not persisted in the RSync status.
+			// TODO: Move requiresRendering out of reconcilerStatus.
+			RequiresRendering: false,
```

Comment:
This can be read from the `RequiresRenderingAnnotationKey` annotation

Reply:
Should we do that here though?

It seems like it's populated by the `RENDERING_ENABLED` env var or `--rendering-enabled` cmd flag, then passed into `parse.Options.RenderingEnabled` which affects control flow in RunFunc which calls `setRequiresRendering` to update the annotation value. None of that seems to check the cached status... 

Rendering seems to behave inconsistently to source and sync, which have `needToSetSourceStatus` and `needToSetSyncStatus` that compare against the cached status for optimization.

Reply:
It looks like it should get set shortly after when files are read from the filesystem, so perhaps it doesn't make a difference.